iothub_client/inc/iothub_client_properties.h

```diff
@@ -142,76 +147,76 @@ typedef struct IOTHUB_CLIENT_DESERIALIZED_PROPERTY_TAG {
 *                                         Note: This is NOT a null-terminated string.
 * @param[out]  serializedPropertiesLength Length of serializedProperties
 * 
 * @remarks  Applications typically will invoke this API when processing a property from service 
 *           (IOTHUB_CLIENT_PROPERTIES_RECEIVED_CALLBACK) to indicate whether properties have been accepted or rejected 
 *           by the device.  For example, if the service requested DesiredTemp=50, then the application could fill out 
-*           the fields in an IOTHUB_CLIENT_WRITABLE_PROPERTY_RESPONSE structure indicating whether the request can be 
+*           the fields in an IOTHUB_CLIENT_PROPERTY_WRITABLE_RESPONSE structure indicating whether the request can be 
 *           processed or whether there was a failure.
 *
 *           Applications can also manually construct the payload based on the convention rules.  This API is an easier 
-*           to use alternative, so the application can use the more convenient IOTHUB_CLIENT_WRITABLE_PROPERTY_RESPONSE 
+*           to use alternative, so the application can use the more convenient IOTHUB_CLIENT_PROPERTY_WRITABLE_RESPONSE 
 *           structure instead of raw serialization.
 *
-*           This API does not perform any network I/O.  It only translates IOTHUB_CLIENT_WRITABLE_PROPERTY_RESPONSE into
+*           This API does not perform any network I/O.  It only translates IOTHUB_CLIENT_PROPERTY_WRITABLE_RESPONSE into
 *           a byte array.  Applications should use APIs such as IoTHubDeviceClient_LL_SendPropertiesAsync to send the 
 *           serialized data.
 *
 * @return   IOTHUB_CLIENT_OK upon success or an error code upon failure.
 */
-MOCKABLE_FUNCTION(, IOTHUB_CLIENT_RESULT, IoTHubClient_Serialize_WritablePropertyResponse, const IOTHUB_CLIENT_WRITABLE_PROPERTY_RESPONSE*, properties, size_t, numProperties, const char*, componentName, unsigned char**, serializedProperties, size_t*, serializedPropertiesLength);
+MOCKABLE_FUNCTION(, IOTHUB_CLIENT_RESULT, IoTHubClient_Properties_Writer_CreateWritableResponse, const IOTHUB_CLIENT_PROPERTY_WRITABLE_RESPONSE*, properties, size_t, numProperties, const char*, componentName, unsigned char**, serializedProperties, size_t*, serializedPropertiesLength);
 
 /**
-* @brief   Frees serialized properties that were initially allocated with IoTHubClient_Serialize_ReportedProperties() 
-*          or IoTHubClient_Serialize_WritablePropertyResponse().
+* @brief   Frees serialized properties that were initially allocated with IoTHubClient_Properties_Writer_CreateReported() 
+*          or IoTHubClient_Properties_Writer_CreateWritableResponse().
 *
 * @param[in]  serializedProperties Properties to free.
 */
-MOCKABLE_FUNCTION(, void, IoTHubClient_Serialize_Properties_Destroy, unsigned char*, serializedProperties);
+MOCKABLE_FUNCTION(, void, IoTHubClient_Properties_Properties_Writer_Destroy, unsigned char*, serializedProperties);
```

Comment:
Are we able to delete one of the `properties`?
```suggestion
MOCKABLE_FUNCTION(, void, IoTHubClient_Properties_Writer_Destroy, unsigned char*, serializedProperties);
```